WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java

```diff
@@ -173,32 +172,8 @@ public static ReaderPost fromJson(JSONObject json) {
         // xpost info
         assignXpostIdsFromJson(post, json.optJSONArray("metadata"));
 
-        // if there's no featured image, check if featured media has been set - this is sometimes
-        // a YouTube or Vimeo video, in which case store it as the featured video so we can treat
-        // it as a video
-        if (!post.hasFeaturedImage()) {
-            JSONObject jsonMedia = json.optJSONObject("featured_media");
-            if (jsonMedia != null && jsonMedia.length() > 0) {
-                String mediaUrl = JSONUtils.getString(jsonMedia, "uri");
-                if (!TextUtils.isEmpty(mediaUrl)) {
-                    String type = JSONUtils.getString(jsonMedia, "type");
-                    boolean isVideo = (type != null && type.equals("video"));
-                    if (isVideo) {
-                        post.featuredVideo = mediaUrl;
-                    } else {
-                        post.featuredImage = mediaUrl;
-                    }
-                }
-            }
-        }
-        // if the post still doesn't have a featured image but we have attachment data, check whether
-        // we can find a suitable featured image from the attachments
-        if (!post.hasFeaturedImage() && post.hasAttachments()) {
-            post.featuredImage = new ImageSizeMap(post.getText(), post.attachmentsJson)
-                    .getLargestImageUrl(ReaderConstants.MIN_FEATURED_IMAGE_WIDTH);
-        }
-        // if we *still* don't have a featured image but the text contains an IMG tag, check whether
-        // we can find a suitable image from the text
+        // if the post doesn't have a featured image but it contains an IMG tag, check whether
+        // we can find a suitable image from the content
         if (!post.hasFeaturedImage() && post.hasImages()) {
             post.featuredImage = new ReaderImageScanner(post.text, post.isPrivate)
```

Comment:
My two cents: 
Blog posts and reader posts share a lot of similarities but a featured image for the reader isn't the same thing as a featured image for a user's blog. Same property names perhaps but the intention is different, and reflective of the terminology used by the reader squad to describe the post and the image shown.
If we were ever to try to reuse reader posts as blog posts I'd agree 100% with the concern about repurposing a property, but then our use case would have changed and the distinction would be proper.  However, in this particular case, maybe we're conflating a property name used by reader posts with blog posts (which is already a type distinction) unnecessarily.  By design a reader post's "featured image" is not necessarily an image selected by the user. 
For that reason I'd leave it as is.